### PR TITLE
Audio: Fix the aligned feature comments

### PR DESCRIPTION
**Pull request by andrula-song**
Fix the comments of frame_align and frame_align_shift to make them more clear and understandable.

### CODE REVIEW

File: src/include/sof/audio/audio_stream.h [hunk 1]
```diff
@@ -46,15 +46,23 @@ struct sof_audio_stream_params {
...
 	uint16_t channels;	/**< Number of samples in each frame */
 
-	/** alignment limit of stream copy, this value indicates how many
-	 * integer frames which can meet both byte align and frame align
-	 * requirement. it should be set in component prepare or param functions
+	/**
+	 * Alignment limit of stream copy, used by audio_stream_avail_frames_aligned.
+	 * We can set byte_align and frame_align in the init or prepare functions to
+	 * limit the processing data size is byte_align bytes aligned and frame_align
+	 * frames aligned. This feature indicates how many integer frames can meet both
```
Comment (btian1) [inline src/include/sof/audio/audio_stream.h:53]: feature or number or data or value is better?

**Review by btian1 — COMMENTED** [review]
Normally, one variable does not need add so much comments, since you are adding, this means this variable is hard to understand directly,  how about add one example to show what exactly it means.

File: src/include/sof/audio/audio_stream.h (hunk 1 above)
Comment (plbossart) [reply src/include/sof/audio/audio_stream.h:53]: that comment is not helpful @andrula-song. It's not clear if 'byte_align' is a boolean or a number of bytes to align on. Likewise 'frame_aligned' could be a boolean but you seem to use it as a counter. 

Consider renaming the structure members to describe exactly what they do in self-explanatory ways.

File: src/include/sof/audio/audio_stream.h [hunk 2]
```diff
@@ -46,15 +46,23 @@ struct sof_audio_stream_params {
...
 	/**
-	 * alignment limit of stream copy, alignment is the frame_align_shift-th
-	 * power of 2 bytes. it should be set in component prepare or param functions
+	 * Alignment limit of stream copy, used by audio_stream_avail_frames_aligned.
+	 * We use right shifting frame_align_shift bits to replace divising framesize to
+	 * avoid runtime division. That means, if the frame size and the frame_align
+	 * we set both are power of 2, we will get the same results by shifting and
+	 * division. Otherwise we will get smaller available frame number than division,
+	 * and processing data size is the frame_align_shift-th power of 2 bytes aligned.
+	 * It should be set in component prepare or param functions
```
Comment (plbossart) [inline src/include/sof/audio/audio_stream.h:65]: same, you are making the explanations more verbose without explaining the concepts.

**Review by plbossart — CHANGES_REQUESTED** [review]

File: src/include/sof/audio/audio_stream.h (hunk 2 above)
Comment (lgirdwood) [reply src/include/sof/audio/audio_stream.h:65]: @andrula-song adding some example context is also helpful to explain. e.g. 
```
....using xtensa HiFi3 then its' best we align frame sizes of X bytes, e.g. a 16 bit stereo stream with Hifi3 would be Y frames and Z bytes....using Hifi4 then ......
```

File: src/include/sof/audio/audio_stream.h [hunk 3]
```diff
@@ -46,17 +46,20 @@ struct sof_audio_stream_params {
...
 	uint16_t channels;	/**< Number of samples in each frame */
 
-	/** alignment limit of stream copy, this value indicates how many
-	 * integer frames which can meet both byte align and frame align
-	 * requirement. it should be set in component prepare or param functions
+	/**
+	 * align_frame_cnt indicates minimum frames can meet both bytes align and
+	 * frames align requirement. Eg: for a stereo 16-bit stream, the processing
+	 * data must be 8 bytes align as well as 3 frames align to meet the xtensa
+	 * intrinsic and algorithm requirement, then the align_frame_cnt is 6.
```
Comment (lgirdwood) [inline src/include/sof/audio/audio_stream.h:53]: Would 16bit stereo not be 4 bytes aligned instead of 8 ?

Comment (andrula-song) [reply src/include/sof/audio/audio_stream.h:53]: the multi-way load/store xtensa intrinsic needs to be 8-byte aligned.

**Review by btian1 — APPROVED** [review]

File: src/include/sof/audio/audio_stream.h (hunk 3 above)
Comment (lgirdwood) [reply src/include/sof/audio/audio_stream.h:53]: > the multi-way load/store xtensa intrinsic needs to be 8-byte aligned.

ok, so the instrinsic requirement is 8 bytes (2 frames) and the algo requiremnet is 3 frames and hence the common factor processing block size is 6 frames.. You need to break it down as follows so it's easier to follow.

e.g. Consider an algorithm that processes in blocks of 3 frames configured to prcess 16bit stereo using xtensa HiFi3 SIMD. Therefore with 16 bit stereo we have a frame size of 4 bytes, and SIMD intrinsic requirment of 2 frames for HiFi X and a an algorithim requirment of 3 frames. Hence the common processing block size has to align with frame(1), intrinsic(2) and algorithm (3) giving us an optimum processing block size of 6 frames.

**Review by lgirdwood — APPROVED** [review]

**Comment by lgirdwood** [comment]
@plbossart good now ?

**Review by singalsu — APPROVED** [review]

File: src/include/sof/audio/audio_stream.h [hunk 4]
```diff
@@ -46,17 +46,24 @@ struct sof_audio_stream_params {
 	uint32_t rate;		/**< Number of data frames per second [Hz] */
 	uint16_t channels;	/**< Number of samples in each frame */
 
-	/** alignment limit of stream copy, this value indicates how many
-	 * integer frames which can meet both byte align and frame align
-	 * requirement. it should be set in component prepare or param functions
+	/**
+	 * align_frame_cnt indicates minimum frames can meet both bytes align and
```
Comment (lyakh) [inline src/include/sof/audio/audio_stream.h:50]: not a native English speaker, but maybe "minimum number of frames, that satisfies both byte and frame alignment requirements."

**Review by lgirdwood — APPROVED** [review]

**Comment by lgirdwood** [comment]
SOFCI TEST

**Review by dbaluta — APPROVED** [review]

**Pull request merged**